File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int freq = 0, res = 0;
        
        for (int num: nums) {
            if (freq == 0) {
                res = num;
            }

            if (num == res) {
                freq++;
            } else {
                freq--;
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        res = []
        queue = deque()
        l = 0

        for r in range(len(nums)):
            while len(queue) and nums[queue[-1]] < nums[r]:
                queue.pop()
            
            queue.append(r)
            window = r - l + 1

            if window == k:
                res.append(nums[queue[0]])
                if queue[0] == l:
                    queue.popleft()
                l += 1
        
        return res